File: src/com/tunitask/entities/Reclamation.java
package com.tunitask.entities;

import com.tunitask.utils.Statics;

public class Reclamation implements Comparable<Reclamation> {

    private int id;
    private User user;
    private String nom;
    private String prenom;
    private String email;
    private String description;
    private String etat;

    public Reclamation() {
    }

    public Reclamation(int id, User user, String nom, String prenom, String email, String description, String etat) {
        this.id = id;
        this.user = user;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.description = description;
        this.etat = etat;
    }
      public Reclamation( String nom, String prenom, String email, String description, String etat) {
        
       
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.description = description;
        this.etat = etat;
    }
      
        public Reclamation( int id,String nom, String prenom, String email, String description, String etat) {
        
       this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.email = email;
        this.description = description;
        this.etat = etat;
    }

   

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }


    @Override
    public int compareTo(Reclamation reclamation) {
        switch (Statics.compareVar) {
            case "User":
                return this.getUser().getEmail().compareTo(reclamation.getUser().getEmail());
            case "Nom":
                return this.getNom().compareTo(reclamation.getNom());
            case "Prenom":
                return this.getPrenom().compareTo(reclamation.getPrenom());
            case "Email":
                return this.getEmail().compareTo(reclamation.getEmail());
            case "Description":
                return this.getDescription().compareTo(reclamation.getDescription());
            case "Etat":
                return this.getEtat().compareTo(reclamation.getEtat());

            default:
                return 0;
        }
    }

}

File: src/Entity/Reclamation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use App\Validator\ContainsBadWords; 
use App\Entity\Users;
use Symfony\Component\Serializer\Annotation\Groups;


#[ORM\Entity]
class Reclamation
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(type: "integer", nullable: false)]
     #[Groups("Reclamations")]
    private $id;

    #[ORM\Column(type: "string", length: 50, nullable: false)]
    #[Assert\NotBlank()]
    #[Groups("Reclamations")]
    private $nom;

    #[ORM\Column(type: "string", length: 50, nullable: false)]
    #[Assert\NotBlank()]
    #[Groups("Reclamations")]
    private $prenom;

    #[ORM\Column(type: "string", length: 50, nullable: false)]
    #[Assert\NotBlank()]
    #[Assert\Email()]
#[Groups("Reclamations")]
    private $email;

    #[ORM\Column(type: "string", length: 255, nullable: false)]
    #[Assert\NotBlank()]
    #[ContainsBadWords]
    #[Groups("Reclamations")]
    private $description;

    #[ORM\Column(type: "string", nullable: false)]
    #[Groups("Reclamations")]
    private $etat;

    #[ORM\ManyToOne(targetEntity: "Users")]
    #[ORM\JoinColumn(name: "id_user", referencedColumnName: "id")]
    #[Groups("Reclamations")]
    private  $idUser;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getIdUser(): ?Users
    {
        return $this->idUser;
    }

    public function setIdUser(?Users $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function __toString()
    {
        return (string) $this->prenom . ' ' . $this->nom . ' | ' . $this->description;
    }
}
